src/components/molecules/Menu/Menu.jsx

```diff
@@ -1,6 +1,23 @@
 import "./Menu.scss";
+import React, { useEffect, useRef } from "react";
+
+const Menu = ({ menu, setIsMenuOpen, openSubMenu }) => {
+
+  const menuRef = useRef();
+  useEffect(() => {
+    const handler = (e) => {
+      if (menuRef.current && !menuRef.current.contains(e.target)) {
+        setIsMenuOpen(false);
+      }
+    };
+
+    document.addEventListener("click", handler);
+
+    return () => {
+      document.removeEventListener("click", handler);
+    };
+  });
 
-const Menu = ({ menu, showMenu, openSubMenu }) => {
   const setMenuContent = (item, i) => {
     if (item.title === "Search") {
       return (
```

Comment:
Why are you wrapping input in article? This seems wrong to me

Reply:
Changed to <div> , is it OK ?